src/main/java/use_case/probability/stand/ProbabilityStandInteractor.java

```diff
@@ -0,0 +1,237 @@
+     * @return a map that contains a key representing how many scenarios have been tested and in how many of them the
+     * player wins.
+     */
+    private Map<String, Integer> countWinsandGames(int userScore, Map<Integer, Integer> currentDealer,
+                                                   Map<Integer, Integer> unknownDeck) {
+
+        int playerWin = 0;
+        int scenarioCount = 0;
+        Map<String, Integer> result = new HashMap<>();
+
+        result.put(SCENARIO, 0);
+        result.put(WINS, 0);
+
+        for (int i = 1; i <= 11; i++) {
+
+            // check if there is at least one card of this value left in the unknown card list, if there is not then
+            // continue to the next value to remove useless iterations.
+            if (unknownDeck.get(i) < 1) {
+                continue;
+            }
+
+            Map<Integer, Integer> conditionalDealer = new HashMap<>(currentDealer);
+            conditionalDealer.compute(i, (k, v) -> v + 1);
+
+
+            int dealerScore = handScore(conditionalDealer);
+
+            if (dealerScore > BLACKJACK) {
+                playerWin += unknownDeck.get(i);
+                scenarioCount += unknownDeck.get(i);
+            }
+
+            // dealerScore will not result in a bust.
+            else if (dealerScore >= 17) {
+
+                if (dealerScore < userScore) {
+                    playerWin += unknownDeck.get(i);
+                }
+                scenarioCount += unknownDeck.get(i);
+            }
+
+            else  {
+
+                Map<Integer, Integer> conditionalUnknown = new HashMap<>(unknownDeck);
+                conditionalUnknown.compute(i, (k, v) -> v - 1);
+
+                // edge-case that the deck is empty to prevent infinite recursion
+                if (this.sumCollection(conditionalUnknown.values()) == 0) {
+                    playerWin++;
+                    scenarioCount++;
+                }
+
+                // calculates all possible scenarios assuming dealer has card in hand and pulls again.
+                // from a non-recursive call to this function the max recursive calls that can be made is 7
+                // assuming the dealer pulls/has four Ones, four twos and one three (score of 15, then there will
+                // one more recursive call).
+                else {
+                    Map<String, Integer> heldResult = this.countWinsandGames(userScore,
+                            conditionalDealer, conditionalUnknown);
+
+                    result.compute(SCENARIO, (k,v) -> v + heldResult.get(SCENARIO));
+                    result.compute(WINS, (k,v) -> v + heldResult.get(WINS));
+                }
+            }
+
+            int finalScenarioCount = scenarioCount;
+            int finalWinCount = playerWin;
+
+            result.compute(SCENARIO, (k, v) -> v + finalScenarioCount);
+            result.compute(WINS, (k, v) -> v + finalWinCount);
+        }
+
+        return result;
```

Comment:
This whole method is really nicely documented and easy to follow even though it's the most substantial part of your algorithm and relatively complicated! Also the use of recursion is really creative and it's great that it's only possible to have seven layers of recursion so it doesn't get too inefficient.